api/types/src/move.rs

```diff
@@ -0,0 +1,412 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Address;
+use move_core_types::{
+    identifier::Identifier,
+    language_storage::{StructTag, TypeTag},
+};
+use resource_viewer::{AnnotatedMoveStruct, AnnotatedMoveValue};
+
+use serde::{Deserialize, Serialize, Serializer};
+use std::{collections::BTreeMap, convert::From, result::Result};
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct MoveResource {
+    #[serde(rename = "type")]
+    pub typ: String,
+    pub type_tag: MoveResourceType,
+    pub value: MoveStructValue,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum MoveResourceType {
+    Struct(MoveStructTag),
+}
+
+impl From<AnnotatedMoveStruct> for MoveResource {
+    fn from(s: AnnotatedMoveStruct) -> Self {
+        Self {
+            typ: s.type_.to_string(),
+            type_tag: MoveResourceType::Struct(MoveStructTag::from(s.type_.clone())),
+            value: MoveStructValue::from(s),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct U64(u64);
+
+impl Serialize for U64 {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        self.0.to_string().serialize(serializer)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct U128(u128);
+
+impl Serialize for U128 {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        self.0.to_string().serialize(serializer)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct HexEncodedBytes(Vec<u8>);
+
+impl Serialize for HexEncodedBytes {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        format!("0x{}", &hex::encode(&self.0)).serialize(serializer)
+    }
+}
```

Comment:
Do you plan to add Deserialize impls for these types so that clients will be able to read them more easily?

Reply:
add deserializer later, this PR only focus on render account resources.